client/src/templates/Challenges/utils/frame.ts

```diff
@@ -273,11 +273,47 @@ const initTestFrame = (frameReady?: () => void) => (frameContext: Context) => {
 const initMainFrame =
   (_: unknown, proxyLogger?: ProxyLogger) => (frameContext: Context) => {
     waitForFrame(frameContext)
-      .then(() => {
+      .then(async () => {
+        const iframeBuild: string = frameContext.build;
+        const parser = new DOMParser();
+        const iframeBuildDoc = parser.parseFromString(iframeBuild, 'text/html');
+        const allLinks = iframeBuildDoc.querySelectorAll(
+          'script[src], link[href]'
+        );
+        let errMsgs = '';
+        let unfetchedLinks = allLinks.length;
+
+        if (!unfetchedLinks) return null;
+
+        for (const url of allLinks) {
+          const urlAddr =
+            url.tagName === 'SCRIPT'
+              ? (url as HTMLScriptElement).src
+              : (url as HTMLLinkElement).href;
+
+          try {
+            const response = await fetch(String(urlAddr));
```

Comment:
```suggestion
            const response = await fetch(url);
```
It's already a string.